lab-tim/__test__/route/route-toy.test.js

```diff
@@ -0,0 +1,206 @@
+          expect(this.errPost.message).toBe('Internal Server Error');
+        });
+        test('should return 404 on invalid endpoint', done => {
+          superagent.post('localhost:3000/bad/endpoint')
+            .type('application/json')
+            .send({})
+            .catch(err => {
+              expect(err.status).toBe(404);
+              done();
+            });
+        });
+      });
+    });
+
+    describe('GET requests', () => {
+      describe('Valid Requests', () => {
+        beforeAll(done => {
+          superagent.get(`localhost:3000/api/toy/${this.mockToy._id}`)
+            .then(res => {
+              this.getToy = res.body;
+              this.resGet = res;
+              done();
+            });
+        });
+        test('should get a toy record, given a valid request', () => {
+          expect(this.getToy).toBeInstanceOf(Object);
+          expect(this.getToy).toHaveProperty('name');
+          expect(this.getToy).toHaveProperty('desc');
+          expect(this.getToy).toHaveProperty('_id');
+        });
+        test('should have an _id, name, and desc, given a valid request', () => {
+          expect(this.getToy._id).toMatch(/^[a-f\d]{24}$/i);
+          expect(this.getToy.name).toBe('slinky');
+          expect(this.getToy.desc).toBe('red plastic');
+        });
+        test('should return a 200 OK, given a valid request', () => {
+          expect(this.resGet.status).toBe(200);
+        });
+      });
+
+      describe('Invalid Requests', () => {
+        beforeAll(done => {
+          superagent.get(`localhost:3000/api/toy/${this.mockToy._id}badId`)
+            .catch(err => {
+              this.errGet = err;
+              done();
+            });
+        });
+        test('should return a status of 400 Bad Request', () => {
+          // expect(this.errGet.status).toBe(400);
+          // expect(this.errGet.message).toBe('Bad Request');
+          expect(this.errGet.status).toBe(500);
+          expect(this.errGet.message).toBe('Internal Server Error');
+        });
+        test('should return 404 on invalid endpoint', done => {
+          superagent.get('localhost:3000/bad/endpoint')
+            .catch(err => {
+              expect(err.status).toBe(404);
+              done();
+            });
+        });
+      });
+    });
+
+    describe('PUT requests', () => {
+      describe('Valid Requests', () => {
+        beforeAll(done => {
+          superagent.put(`localhost:3000/api/toy/${this.mockToy._id}`)
+            .type('application/json')
+            .send({
+              name: 'slinky',
+              desc: 'blue plastic'
+            })
+            .then(res => {
+              this.resPut = res;
+              done();
+            });
+        });
+        test('should return a 204 NO CONTENT, given a valid request', () => {
+          expect(this.resPut.status).toBe(204);
+        });
+      });
+
+      describe('Invalid Requests', () => {
+        beforeAll(done => {
```

Comment:
My guess is that your tests failing has something to do with this beforeAll block. I don't think it is actually throwing an error without content being sent. I think mongo reads an empty object as a valid input. If I have some more time I will keep messing with it but with the refactor to the error module in the next lab you might have it working by then. It might be a good question to ask Scott.